functorch/csrc/BatchRulesPooling.cpp

```diff
@@ -82,6 +82,46 @@ max_pool2d_with_indices_batch_rule(
       reshape_dim_outof(0, bdim_size, std::get<1>(result)), 0);
 }
 
+// We can't use ALL_TENSORS_HAVE_OPTIONAL_BDIM_BOXED because the CUDA
```

Comment:
Perhaps it's better to just augment the boxed rule with the option to make some args contiguous?

Reply:
@Chillee good idea! I did an implementation with one tensor that could be contiguous (later we can make it more generic if needed). What do you think ?

Reply:
Looks good to me!